src/chains/ethereum/transaction/tests/index.test.ts

```diff
@@ -1,6 +1,317 @@
+          txFromRpc.s.toBuffer()
+        );
+        const txFromDb = TransactionFactory.fromDatabaseTx(db, common);
+        assert.strictEqual(txFromDb.type.toString(), "0x0");
+      });
+      it("generates legacy transactions from raw string", async () => {
+        const str: string = "0x" + txFromRpc.serialized.toString("hex");
+        const txFromString = TransactionFactory.fromString(str, common);
+        assert.strictEqual(txFromString.type.toString(), "0x0");
+      });
+    });
+
+    describe("EIP2930AccessListTransaction type from factory", () => {
+      let txFromRpc: EIP2930AccessListTransaction;
+      let key: string;
+      const accessListStorageKey =
+        "0x0000000000000000000000000000000000000000000000000000000000000004";
+
+      it("generates eip2930 access list transactions from rpc data", async () => {
+        const rpc: TypedRpcTransaction = {
+          from: from,
+          to: to,
+          type: "0x1",
+          gasPrice: "0xffff",
+          accessList: [
+            {
+              address: accessListAcc,
+              storageKeys: [accessListStorageKey]
+            }
+          ]
+        };
+        txFromRpc = <EIP2930AccessListTransaction>(
+          TransactionFactory.fromRpc(rpc, common)
+        );
+        txFromRpc.signAndHash(fakePrivateKey);
+        key = txFromRpc.accessListJSON[0].storageKeys[0];
+        assert.strictEqual(txFromRpc.type.toString(), "0x1");
+        assert.strictEqual(key, accessListStorageKey);
+      });
+      it("generates eip2930 access list transactions from raw buffer data", async () => {
+        const db: TypedDatabaseTransaction = txFromRpc.toEthRawTransaction(
+          txFromRpc.v.toBuffer(),
+          txFromRpc.r.toBuffer(),
+          txFromRpc.s.toBuffer()
+        );
+        const txFromDb = <EIP2930AccessListTransaction>(
+          TransactionFactory.fromDatabaseTx(db, common)
+        );
+        key = txFromDb.accessListJSON[0].storageKeys[0];
+        assert.strictEqual(txFromDb.type.toString(), "0x1");
+        assert.strictEqual(key, accessListStorageKey);
+      });
+
+      it("generates eip2930 access list transactions from raw string", async () => {
+        const str: string = "0x" + txFromRpc.serialized.toString("hex");
+        const txFromString = <EIP2930AccessListTransaction>(
+          TransactionFactory.fromString(str, common)
+        );
+        key = txFromString.accessListJSON[0].storageKeys[0];
+        assert.strictEqual(txFromString.type.toString(), "0x1");
+        assert.strictEqual(key, accessListStorageKey);
+      });
+    });
+  });
+
+  describe("LegacyTransaction Type", () => {
+    const rpc: TypedRpcTransaction = {
+      from: from,
+      to: to,
+      type: "0x0",
+      gasPrice: "0xffff"
+    };
+    const tx = <LegacyTransaction>TransactionFactory.fromRpc(rpc, common);
+    it("can be signed and hashed", () => {
+      // TODO: can we know the hash ahead of time if we supply the right data?
```

Comment:
see above comment about deterministic accounts.

Reply:
Updated to have deterministic accounts